File: typescript/src/points/polarpoint.ts
interface PolarPoint extends Point {
    len: number,
    angle: number,
    computeX: () => number
    computeY: () => number
}

function newPolarPoint (len: number, angle: number): PolarPoint {
    return {
        len,
        angle,

        computeX: function (): number {
            return Math.round(len * Math.cos(angle))
        },

        computeY: function (): number {
            return Math.round(len * Math.sin(angle))
        },

        getX: function () : number {
            return this.computeX();
        },

        getY: function () : number {
            return this.computeY();
        }
    }
}

export { PolarPoint, newPolarPoint }

File: java/src/main/java/edu/cmu/cs/cs214/rec02/points/PolarPoint.java
package edu.cmu.cs.cs214.rec02.points;

public class PolarPoint implements Point {
    private double len;
    private double angle;

    public PolarPoint (double len, double angle) {
        this.len = len;
        this.angle = angle;
    }

    private int computeX() {
        return (int) (this.len * Math.cos(this.angle));
    }

    private int computeY() {
        return (int) (this.len * Math.sin(this.angle));
    }

    public int getX() {
        return computeX();
    }

    public int getY() {
        return computeY();
    }
}
